grid2op/Reward/_alertTrustScore.py

```diff
@@ -0,0 +1,165 @@
+        #self.alert_attack_blackout = 0
+        #self.no_alert_attack_no_blackout = 0
+        #self.no_alert_attack_blackout = 0
+        #self.blackout_encountered = False
+            
+    def __call__(self, action, env, has_error, is_done, is_illegal, is_ambiguous):
+
+        score_ep = 0.
+        if self._is_simul_env:
+            return score_ep
+
+        self.blackout_encountered = self.is_in_blackout(has_error, is_done)
+        
+        score_ep = 0.
+        if self._is_simul_env:
+            return score_ep
+            
+        res = super().__call__(action, env, has_error, is_done, is_illegal, is_ambiguous)
+        self.cumulated_reward += res
+
+        is_attack = (env._time_since_last_attack == 0).any()#even if there are simultaneous attacks, we consider this as a single attack event
+        self.total_nb_attacks += is_attack
+
+        # TODO
+        #lines_alerted_beforeattack = np.equal(env._time_since_last_alert, env._time_since_last_attack + 1) and lines_attacked
+        #self.total_nb_alerts += np.sum(lines_alerted_beforeattack)
+            
+        if not is_done:
+
+            # TODO
+            #lines_attacked_no_blackout = env._time_since_last_attack == SURVIVOR_TIMESTEPS
+#
+            #self.alert_attack_no_blackout += np.sum(lines_alerted_beforeattack[lines_attacked_no_blackout])
+            #self.no_alert_attack_no_blackout += np.sum(~lines_alerted_beforeattack[lines_attacked_no_blackout])
+            
+            return score_ep
+            
+        else:
+            self.nb_last_attacks=np.sum(np.any(self._ts_attack,axis=1))
+            cm_reward_min_ep, cm_reward_max_ep = self._compute_min_max_reward(self.total_nb_attacks,self.nb_last_attacks)
+            score_ep = self._normalisation_fun(self.cumulated_reward, cm_reward_min_ep, cm_reward_max_ep,self.min_score,self.max_score,self.blackout_encountered)
+
+            # TODO
+            #if self.blackout_encountered:
+            #    lines_attacked_dangerzone = (env._time_since_last_attack >= 0) * (env._time_since_last_attack < SURVIVOR_TIMESTEPS)
+            #
+            #    self.alert_attack_blackout += 1. * any(lines_alerted_beforeattack[lines_attacked_dangerzone])
+            #    self.no_alert_attack_blackout += 1. * any(~lines_alerted_beforeattack[lines_attacked_dangerzone])
+            #else :
+            #    lines_attacked_no_blackout = env._time_since_last_attack > 0
+            #
+            #    self.alert_attack_no_blackout += np.sum(lines_alerted_beforeattack[lines_attacked_no_blackout])
+            #    self.no_alert_attack_no_blackout += np.sum(~lines_alerted_beforeattack[lines_attacked_no_blackout])
+            
+            return score_ep
+        
+    @staticmethod
+    def _normalisation_fun(cm_reward, cm_reward_min_ep, cm_reward_max_ep,min_score,max_score,is_blackout,tol=1e-5):
+        standardized_score = np.round((cm_reward - cm_reward_min_ep) / (cm_reward_max_ep - cm_reward_min_ep +tol),4)
```

Comment:
Previous answer "Because then you get score of 0.99993.
Where should we round this then ? Here it is to compensate for the addition of the tolerance that is artificial"

New answer: I remove the tol argument and rounding. cm_reward_max_ep == cm_reward_min_ep is a case that we deal with before, and don't call this _normalisation_fun if that's the case